src/ansys/templates/python/solution/{{cookiecutter.__project_name_slug}}/.github/workflows/build-release.yml

```diff
@@ -65,19 +67,137 @@ jobs:
+        with:
+          sbom-artifact: {{ "${{ env.SBOM_NAME }}" }}
+          gh-token: {{ "${{ secrets.WORKFLOW_TOKEN }}" }}
+
   release:
     if: |
       github.event_name == 'release'
     name: Release
-    needs: [check-copyright, check-code-style, build]
+    needs: [build-doc, check-copyright, check-code-style, build, run-tests, check-licenses, package-solution, generate-sbom, validate-sbom]
     runs-on: [ubuntu-latest]
     timeout-minutes: 10
     steps:
       - name: Release
-        uses: Solution-Applications/actions/release@v5
+        uses: ansys-internal/solution-applications-actions/release-to-private-pypi@v11
         with:
           python-version: {{ "${{ env.MAIN_PYTHON_VERSION }}" }}
+          twine-repository-url: https://pkgs.dev.azure.com/pyansys/_packaging/ansys-solutions/pypi/upload/
+          twine-username: "TOKEN"
+          twine-password: {{ "${{ secrets.SOLUTIONS_PRIVATE_PYPI_ADMIN_TOKEN }}" }}
+
+  check-doc-style:
+    name: Check documentation style
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Check documentation style
+        uses: ansys-internal/solution-applications-actions/check-doc-style@v11
+        with:
+          gh-token: {{ "${{ secrets.GITHUB_TOKEN }}" }}
+
+  build-doc:
+    name: Build documentation
+    runs-on: ubuntu-latest
+    needs: check-doc-style
+    timeout-minutes: 10
+    steps:
+      - name: Build documentation
+        uses: ansys-internal/solution-applications-actions/build-doc@v11
+        with:
           gh-token: {{ "${{ secrets.WORKFLOW_TOKEN }}" }}
-          solutions-pypi-token:
-            {{ "${{ secrets.SOLUTIONS_PRIVATE_PYPI_ADMIN_TOKEN }}" }}
-          solutions-pypi-repository-url: https://pkgs.dev.azure.com/pyansys/_packaging/ansys-solutions/pypi/upload/
+          python-version: {{ "${{ env.MAIN_PYTHON_VERSION }}" }}
+          documentation-artifact: {{ "${{ env.HTML_DOCUMENTATION_ARTIFACT}}" }}
+
+  doc-deploy-dev:
+    name: "Deploy development documentation"
+    # Deploy development only when merging or pushing to the 'main' branch
+    if: github.event_name == 'push' && !contains(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    needs: build-doc
+    steps:
+      - uses: ansys/actions/doc-deploy-dev@v5
+        with:
+          doc-artifact-name: {{ "${{ env.HTML_DOCUMENTATION_ARTIFACT}}" }}
+          cname: {{ "${{ env.DOCUMENTATION_CNAME }}" }}
+          token: {{ "${{ secrets.GITHUB_TOKEN }}" }}
+
+  doc-deploy-stable:
+    name: "Deploy stable documentation"
+    # Deploy release documentation when creating a new tag
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    needs: [build-doc, release]
+    steps:
+      - uses: ansys/actions/doc-deploy-stable@v5
+        with:
+          doc-artifact-name: {{ "${{ env.HTML_DOCUMENTATION_ARTIFACT}}" }}
+          cname: {{ "${{ env.DOCUMENTATION_CNAME }}" }}
```

Comment:
I think this ENV var needs to be defined. It may be tricky because I don't think we know this value until the repo is created.

Reply:
AFAIK, this value can't be defined without a page deployment with a random cname from GitHub or a custom one.

Reply:
This is a required input for the action - I'm not sure it makes sense to prompt users for this value when creating the solution. Instead, it may be easier to simply add a hard-coded value for `DOCUMENTATION_CNAME` like `replace.me.com` or similar and just let the action fail by default.

We need to document this, but I think failing the action is a reasonable way to force users to manually populate the correct value.

Reply:
Understood. I put a dummy cname and also added a short comment.